common/functions/src/aggregates/aggregate_stddev_pop.rs

```diff
@@ -0,0 +1,211 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::alloc::Layout;
+use std::fmt;
+use std::marker::PhantomData;
+use std::sync::Arc;
+
+use common_datavalues::prelude::*;
+use common_exception::ErrorCode;
+use common_exception::Result;
+use common_io::prelude::*;
+use num::cast::AsPrimitive;
+
+use super::StateAddr;
+use crate::aggregates::aggregator_common::assert_unary_arguments;
+use crate::aggregates::AggregateFunction;
+use crate::aggregates::AggregateFunctionRef;
+use crate::dispatch_numeric_types;
+
+struct AggregateStddevPopState {
+    pub sum: f64,
+    pub count: u64,
+    pub variance: f64,
+}
+
+impl AggregateStddevPopState {
+    #[inline(always)]
+    fn add(&mut self, value: f64) {
+        self.sum += value;
+        self.count += 1;
+        if self.count > 1 {
+            let t = self.count as f64 * value - self.sum;
+            self.variance += (t * t) / (self.count * (self.count - 1)) as f64;
+        }
+    }
+
+    #[inline(always)]
+    fn merge(&mut self, other: &Self) {
+        if other.count == 0 {
+            return;
+        }
+        if self.count == 0 {
+            self.count = other.count;
+            self.sum = other.sum;
+            self.variance = other.variance;
+            return;
+        }
+
+        let t = (other.count as f64 / self.count as f64) * self.sum - other.sum;
+        self.variance += other.variance
+            + ((self.count as f64 / other.count as f64) / (self.count as f64 + other.count as f64))
+                * t
+                * t;
+        self.count += other.count;
+        self.sum += other.sum;
+    }
+}
+
+#[derive(Clone)]
+pub struct AggregateStddevPopFunction<T> {
+    display_name: String,
+    arguments: Vec<DataField>,
+    t: PhantomData<T>,
+}
+
+impl<T> AggregateFunction for AggregateStddevPopFunction<T>
+where T: DFPrimitiveType + AsPrimitive<f64>
+{
+    fn name(&self) -> &str {
+        "AggregateStddevPopFunction"
+    }
+
+    fn return_type(&self) -> Result<DataType> {
+        Ok(DataType::Float64)
+    }
+
+    fn nullable(&self, _input_schema: &DataSchema) -> Result<bool> {
+        Ok(false)
+    }
+
+    fn init_state(&self, place: StateAddr) {
+        place.write(|| AggregateStddevPopState {
+            sum: 0.0,
+            count: 0,
+            variance: 0.0,
+        });
+    }
+
+    fn state_layout(&self) -> Layout {
+        Layout::new::<AggregateStddevPopState>()
+    }
+
+    fn accumulate(&self, place: StateAddr, arrays: &[Series], _input_rows: usize) -> Result<()> {
+        let state = place.get::<AggregateStddevPopState>();
+        let array: &DFPrimitiveArray<T> = arrays[0].static_cast();
+        for (_row, value) in array.into_no_null_iter().enumerate() {
```

Comment:
Null is really annoying, but we have to take it into consideration.

if array.null_count() == 0 {
   // fastpath
} else {
  array.iter().map...
}

Reply:
ok, I have modified it. PTAL @sundy-li